src/WorldStorage/WSSAnvil.cpp

```diff
@@ -1490,11 +1490,26 @@ bool cWSSAnvil::CheckEntityType(const char * a_IDTag, size_t a_IDTagLength, cons
 
 void cWSSAnvil::LoadEntityFromNBT(cEntityList & a_Entities, const cParsedNBT & a_NBT, int a_EntityTagIdx, const char * a_IDTag, size_t a_IDTagLength)
 {
+	static const EntityLoaderMap EntityTypeToFunction
+	{
+		{ "Boat", &cWSSAnvil::LoadBoatFromNBT },
+		{ "minecraft:boat", &cWSSAnvil::LoadBoatFromNBT }
+
+	};
+
+	auto it = EntityTypeToFunction.find(a_IDTag);
+	if (it != EntityTypeToFunction.end())
+	{
+		(this->*it->second)(a_Entities, a_NBT, a_EntityTagIdx);
+	}
```

Comment:
Created the map and added boat. I do something wrong.
If I load the flat_world from vanilla, the boat is spawned. Restarted the server and the boat is not spawned anymore. The boat was saved a `Boat` in nbt, confirmed that.
Placed then a sheep and a boat in a hole, the sheep is spawned after restart.